front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormModal.tsx

```diff
@@ -25,8 +32,8 @@ const RollingStockEditorFormModal = ({
   const { closeModal } = useModal();
   const { t } = useTranslation(['translation', 'rollingstock']);
 
-  const displayErrorObject = (errorList: RollingStockUsage['usage'][]): JSX.Element => {
-    const projectList = groupBy(errorList, 'project_name');
+  const displayErrorObject = (errorList: RollingStockIsUsed): JSX.Element => {
+    const projectList = groupBy(errorList.RollingStockIsUsed, 'project_name');
```

Comment:
You can test if the rollingStockError is a RollingStockIsUsed error like this, this way no need to use a cast
```suggestion
  const displayErrorObject = (errorList: RollingStockError) => {
    if (typeof errorList !== 'object' || !('RollingStockIsUsed' in errorList)) {
      return null;
    }
    const projectList = groupBy(errorList.RollingStockIsUsed, 'project_name');
```